pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -269,6 +269,11 @@ def should_reset_tensors(self, fx: str) -> bool:
         return is_different_fx and is_first_batch
 
     def reset(self, metrics: Optional[bool] = None) -> None:
+        if self.trainer.sanity_checking:
+            # reset metrics
+            self._progress_bar_metrics = {}
+            self._logged_metrics = {}
+            self._callback_metrics = {}
```

Comment:
Two issues here:

1. We already call this function in `_run_sanity_check`. Instead of adding an `if` here, I woud've performed these calls there directly.

2. Why do we even need to recreate these? Wouldn't it be better to avoid logging anything in the first place during sanity check

Reply:
For 2) the sanity check is to make sure everything runs, so we should also internally record all the logs to make sure the user doesn't run into any logging errors. the only thing we would want to avoid is sending the logs to the logger object for saving.